### ISSUE TITLE
Role access for Content Parts or fields

### ISSUE THREAD

**Issue by JoshTango** [OrchardCMS/OrchardCore#13245]
Configure a content part or field to be accessible by role.

Here is my use case.  I have a site with a content type called channel.  When people make a channel it uses autoroute to create them a route like so: mysite.com/channel/123456789
However some people don't want this number and want something like mysite.com/channel/brandname.  That's obviously  what the alias partis is for.  However I have a project that doesn't want people to be able to be able to decide the alias on their own.

Instead authors will have to fill out a request form saying what alias they would like.  Then an employee would have to do some research and make a decision whether to grant them this alias.

If they get approval it would be nice if an employee with a specific role(not the author) could go into the content item for channels and fill out the alias part.

Another possibility would be to make parts hidden and use some workflow to change a part value.  But I think role based is best.

**Comment by JoshTango** [comment]
I suppose I could just create a module for this with a special permission.  The module would allow an employee to find the content item in question and enter a new route in a form field.  A submit button could trigger some c# that changes the autoroute value.

**Comment by sebastienros** [comment]
> That's what the alias partis is for.

No, that what the autoroute part is for. The alias part is to provide a custom "name" a logical identifier to a content item, not a "route". If you set a custom alias you can't access it via the url with this name, only autoroute can do that. And there is a single autoroute per content item.

I would look into something like Placement to hide a part dynamically, in your case based on roles. Try to create a display handler for the autoroute part shape editor, and would return nothing for specificy roles. Check that when nothing is displayed, then the value is not removed when the content item is updated.
Another simpler option is to have a content handler with a custom EditAsync implementation that will throw an exception if the autoroute value is changed and the user is not in specific roles.
